src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -22,13 +22,85 @@ public sealed class TelemetryConfiguration : IDisposable
         private string instrumentationKey = string.Empty;
         private bool disableTelemetry = false;
         private TelemetryProcessorChainBuilder builder;
         private SnapshottingList<IMetricProcessor> metricProcessors = new SnapshottingList<IMetricProcessor>();
 
         /// <summary>
+        /// Indicates if this instance has been disposed of.
+        /// </summary>
+        private bool isDisposed = false;
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        public TelemetryConfiguration() : this(new InMemoryChannel())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        /// <param name="instrumentationKey">The instrumentation key this configuration instance will provide.</param>
+        public TelemetryConfiguration(string instrumentationKey) : this(new InMemoryChannel())
+        {
+            if (instrumentationKey == null)
+            {
+                throw new ArgumentNullException("instrumentationKey");
+            }
+
+            this.instrumentationKey = instrumentationKey;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        /// <param name="channel">The telemetry channel to provide with this configuration instance.</param>
+        public TelemetryConfiguration(ITelemetryChannel channel)
+        {
+            if (channel == null)
+            {
+                throw new ArgumentNullException("channel");
+            }
+
+            this.telemetryChannel = channel;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        /// <param name="instrumentationKey">The instrumentation key this configuration instance will provide.</param>
+        /// <param name="channel">The telemetry channel to provide with this configuration instance.</param>
+        public TelemetryConfiguration(string instrumentationKey, ITelemetryChannel channel)
+        {
+            if (channel == null)
+            {
+                throw new ArgumentNullException("channel");
+            }
+
+            if (instrumentationKey == null)
+            {
+                throw new ArgumentNullException("instrumentationKey");
+            }
+
+            this.instrumentationKey = instrumentationKey;
+            this.telemetryChannel = channel;
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the TelemetryConfiguration class.
+        /// Frees resources when a TelemetryConfiguration instance is garbage collected if it was not properly disposed of.
+        /// </summary>
+        ~TelemetryConfiguration()
+        {
+            // Cleanup managed resources if the instance was not properly disposed of.
+            this.Dispose(true);
```

Comment:
Per our discussion yesterday I'm removing this finalizer hack and it will be rightly up to the customer not to leak objects by not disposing of them.  Nothing would have broken but since we all agree it's the wrong "solution" anyway this isn't relevant.